### ISSUE TITLE
LCOW: daemon doesn't start with setting experimental to false and a linux layer has been pull

### ISSUE THREAD

**Issue by ebriney** [moby/moby#36560]
**Description**

When a foreign layer is present in image db, the daemon refuses to start with
`fatal: Error starting daemon: operating system is not supported`

**Steps to reproduce the issue:**
1. `docker pull --platform=linux busybox` with `experimental=true`
2. set experimental to false and restarts the daemon
3. daemon fails to start

**Describe the results you received:**
Got `fatal: Error starting daemon: operating system is not supported`

**Describe the results you expected:**
Daemon starts but ignore foreign layers, displays a warning

**Output of `docker version`:**

```
Client:
 Version:       18.03.0-ce-rc1
 API version:   1.37
 Go version:    go1.9.4
 Git commit:    c160c73
 Built: Thu Feb 22 02:34:04 2018
 OS/Arch:       windows/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.03.0-ce-rc1
  API version:  1.37 (minimum version 1.24)
  Go version:   go1.9.4
  Git commit:   c160c73
  Built:        Thu Feb 22 02:41:20 2018
  OS/Arch:      windows/amd64
  Experimental: true
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 18.03.0-ce-rc1
Storage Driver: windowsfilter (windows) lcow (linux)
 Windows:
 LCOW:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 17115 (17115.1.amd64fre.rs4_release.180302-1642)
Operating System: Windows 10 Pro Insider Preview
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 15.89GiB
Name: nuc-win14393
ID: G3IW:EZCY:5TQ2:SAHJ:J3CD:FYNS:2XBK:VGT4:DBK6:5VJO:CLGV:WATK
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 21
 System Time: 2018-03-12T14:11:18.1505413+01:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by ebriney** [comment]
ping @jhowardmsft @johnstep @salah-khan

**Comment by jhowardmsft** [comment]
Yes, this is known. There is a fix in https://github.com/Microsoft/docker/blob/53bdc732ea1af4f97cbb15019c359760de9eb277/image/store.go#L77-L79 (PR https://github.com/moby/moby/pull/36418)

**Comment by ebriney** [comment]
Thanks @jhowardmsft . So for d4w 18.03 stable release we added a message to tell users to reset to default if they want to switch experimental to false. I hope it will be enough.

**Comment by ebriney** [comment]
Fixed in 18.06, I close the issue
